beeHealthy/src/main/java/br/com/devcapu/bee_healthy/application/UserInfoResourceProvider.java

```diff
@@ -64,4 +64,28 @@ public Objective getObjectiveWithText(String selectedObjective) {
     public List<String> getObjectivesValues() {
         return unmodifiableList(objectivesValues);
     }
+
+    public String getString(Gender gender) {
+        for (Map.Entry<String, Gender> entry : genderOptions.entrySet()) {
+            if (gender.equals(entry.getValue())) return entry.getKey();
+        }
+
+        return null;
+    }
+
+    public String getString(ActivityLevel activityLevel) {
+        for (Map.Entry<String, ActivityLevel> entry : activityCoefficientOptions.entrySet()) {
+            if (activityLevel.equals(entry.getValue())) return entry.getKey();
+        }
+
+        return null;
+    }
+
+    public String getString(Objective objective) {
+        for (Map.Entry<String, Objective> entry : objectiveOptions.entrySet()) {
+            if (objective.equals(entry.getValue())) return entry.getKey();
+        }
+
+        return null;
+    }
```

Comment:
Achei `getString` um nome muito genérico. Acho que dá pra ser mais específico.

E dá pra extrair o código comum em um método auxiliar.

```suggestion
    public String getGenderText(Gender gender) {
        return getTextFromValue(genderOptions, gender);
    }

    public String getActivityLevelText(ActivityLevel activityLevel) {
        return getTextFromValue(activityCoefficientOptions, activityLevel);
    }

    public String getObjectiveText(Objective objective) {
        return getTextFromValue(objectiveOptions, objective);
    }

    private <T> String  getTextFromValue(Map<String, T> map, T value) {
        for (Map.Entry<String, T> entry : map.entrySet()) {
            if (value.equals(entry.getValue())) return entry.getKey();
        }
        return null;
    }

```